[Egv2] Update TypeSpec For Warnings

l0lawrence opened the pull request:
--unknown = any
-- cloudevent body to a model

mikeharder committed:
[TypeSpec CI] Add EventGrid

l0lawrence committed:
Merge branch 'typespec-ci-eventgrid' of https://github.com/mikeharder/azure-rest-api-specs into typespec_update

l0lawrence committed:
update format

l0lawrence committed:
regen json but edited to fix autorest bugs

l0lawrence committed:
supress warning and reasoning

l0lawrence committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into typespec_update

l0lawrence committed:
update for flattening object workaround

lmazuel commented on specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:
```diff
@@ -161,131 +165,129 @@ namespace Azure.Messaging.EventGrid {
+    {
+      @doc("Max Events count to be received. Minimum value is 1, while maximum value is 100 events. If not specified, the default value is 1.")
+      @query
+      maxEvents?: int32 = 1;
```
You can define min and max value
https://microsoft.github.io/typespec/standard-library/built-in-decorators#@maxValue

mikeharder committed:
[TypeSpec] Remove "./" from tspconfig.yaml (#24199)

- Prefix is unnecessary and clutters config

mikeharder committed:
Add Messaging.EventGrid back to typespec-ci.yml

msyyc committed:
Fix sqlvm.json (#24248)

azaslonov committed:
Added Azure API Center resource manager. (#24130)

* Added Azure API Center resource manager.

* Added managed identity properties. Changed top-level resource type.

---------

Co-authored-by: Alexander Zaslonov <alzaslon@microsoft.com>

mikeharder committed:
[typespec-ci.yml] Add branch "typespec-next" to PR trigger (#24266)

msyyc committed:
fix (#24250)

TracyYaoYao committed:
[CDN][2023-05-01] Fix async operation options for migrate related APIs (#24164)

Co-authored-by: yaoshi <yaoshi@microsoft.com>

Commit:
Run notebook sdk change (#24269)

* run notebook api

* fix

* fix

piyushdubey committed:
SyMS Service API Swagger Client [2023] (#23180)

* SyMS Service API Swagger Client Spec

* Add SyMS specific custom words + readme

* SyMS: Fix LintDiff errors

* SyMS: Fix LintDiff and Model Validation errors

* SyMS: Fix Model Validation errors

* SyMS: Fix Model Validation errors

* SyMS: Fix Model Validation errors

* SyMS: Suppress Model Validation Errors

* SyMS: Fix Model Validation errors

* SyMS: Fix Model Validation errors

* SyMS: Fix Model Validation errors

* SyMS: Remove Model Validation Errors suppression. Not needed anymore

* SyMS: Move file to latest SDK version

chcmsft committed:
[Maps] Data Registry service GA version (#24204)

* add GA version data registry api

* fix pipeline fails

* update the description of List API

tristanang committed:
Updating ContainerService event grid event schema to support new events (#24105)

* update swagger definition to add new event types

* add examples

* better description

* x-ms-examples

* change to isOutOfSupport

* update to enum

* update to rollingProgressState

* unroll events

* examples

* more examples

* create base prop

* fix object

JoshLove-msft committed:
Fix naming for base models (#24279)

thophan-microsoft committed:
Extend timeZoneId description (#24261)

* Extend timeZoneId property

* Fix ModelValidation errors

* Fix LRO_RESPONSE_HEADER errors

msyyc committed:
Update readme.python.md (#24285)

kevxms committed:
rename cleaup policy type from 'Compaction' to 'Compact' (#24056)

'Compact' is already implemented in the resource provider and is consistent with using verb such as type 'Delete'

prchalla1 committed:
Update Error Code to string (#24150)

olkroshk committed:
Updated the 2021-06-01-preview attestation defintions and fixed (#24124)

versions:

- Updated `2021-06-01-preview/attestation.json` to match the internal swagger defintions.
- Fixed versions in the examples

Squashed commit of the following:

commit 51d313714895a75e734f4b8e37d096d691d3829c
Author: Olga Kroshkina <olkroshk@microsoft.com>
Date:   Tue Oct 4 16:01:55 2022 -0700

    [Public/Global - Microsoft Azure Attestation Service] Commit # 3 - Fixed issues in the public API, added TPM Attestation auth switch

commit ab07e7b296d04ba939d2a1a2a5684da9fb6da8c9
Author: Olga Kroshkina <olkroshk@microsoft.com>
Date:   Tue Oct 4 15:19:42 2022 -0700

    [Public/Global - Microsoft Azure Attestation Service] Commit # 2 - Updated the API version with the 2021-06-01 value

commit 96bd39ed522b49997399492ab4e111fcfe5af3c7
Author: Olga Kroshkina <olkroshk@microsoft.com>
Date:   Tue Oct 4 15:05:43 2022 -0700

    [Public/Global - Microsoft Azure Attestation Service] Commit # 1 - Copy of preview/2021-06-01-preview

olkroshk committed:
[Microsoft Azure Attestation] Control Plane 2021-06-01 (#24125)

* [Public/Global - Microsoft Azure Attestation Service] Commit # 1 - Copy of preview/2021-06-01-preview

* [Public/Global - Microsoft Azure Attestation Service] Commit # 2 - Updated the API version with the 2021-06-01 value

* [Public/Global - Microsoft Azure Attestation Service] Commit # 3 - Fixed issues in the public API, added TPM Attestation auth switch

markcowl committed:
Update Fleet.Management for object deprecation (#24308)

mikeharder committed:
[TypeSpec] Suppress "documentation-required" lint rule (#24297)

- Docs missing from version enums will be fixed by https://github.com/Azure/typespec-azure/issues/3107
- Other missing docs MUST be fixed in the next update to the spec

chandrikagonuguntla committed:
Adding new api version 2023-04-01 for recoveryservicessiterecovery microsoft.recovery services (#24220)

* Adds base for updating Microsoft.RecoveryServices from version stable/2023-02-01 to version 2023-04-01

* Updates readme

* Updates API version in new specs and examples

* Removing blank line

pathuang-sia committed:
s360 fix from https://github.com/Azure/azure-rest-api-specs-pr/pull/12609 (#24222)

Co-authored-by: Patricia Huang <pathuang@pathuang-dev>

msyyc committed:
update python config (#24310)

alluri02 committed:
Fix Swagger Lint issue Organization_Delete.json (#24288)

* Fix Swagger Lint issue Organization_Delete.json

* fix prettier issue

---------

Co-authored-by: Srinivas Alluri <sralluri@microsoft.com>

aim-for-better committed:
Update hdinsight readme tag (#24270)

Co-authored-by: Zhenyu Zhou <zhezhou@microsoft.com>

mikehang-msft committed:
Added GA Specification for ACS Rooms (#24262)

jingjlii committed:
[Hub Generated] Review request for Microsoft.Batch to add version stable/2023-05-01.17.0 (#24181)

* Adds base for updating Microsoft.Batch from version stable/2022-10-01.16.0 to version 2023-05-01.17.0

* Updates readme

* Updates API version in new specs and examples

* Update examples and spec for new API version.

hstellmach-msft committed:
Azure OpenAI 2023-06-01-preview (#24252)

* Azure OpenAI: Add 2023-06-01-preview version

* Update example for get_images_operation

* Fix FT examples

mikeharder committed:
[AI.TextTranslation] Disambiguate example titles (#24323)

- Causes error in TypeSpec 0.45.0
- Also change casing of textType in example to match spec
  - Fixes error ENUM_CASE_MISMATCH in Swagger ModelValidation

rheajain committed:
Add new stable version for PaloAltoNetworks (#24005)

* Add new stable version for PaloAltoNetworks

* change version in readme

* Add API version tag

* Rectify Certificate and Firewalls example Ids

Liran-E committed:
[Hub Generated] Review request for Microsoft.AlertsManagement to add version stable/2019-03-01 (#24110)

* change api version to the proper api version

* changed reference to common types api-version

* changed TargetTypeParameter to type method

---------

Co-authored-by: MIDDLEEAST\lienglender <lienglender@microsoft.com>

raych1 committed:
Added a wrapper script for writer to generate sdk (#24267)

* Added a wrapper script for writer to generate sdk

* output sdk repo root path

* Set default value for typespec project folder.

ismael-ms committed:
Removed DEM deprecation banner as it has now been deprecated (#24038)

Merging. Notes - 

1. The enum values in the API 2022-08-01 version are deprecated - approved by the Azure retirement team.
2. While the API version is stable, non of the SDKs are GA-ed; [all](https://learn.microsoft.com/en-us/azure/azure-maps/rest-sdk-developer-guide) are shipped only in preview mode.

l0lawrence committed:
enum doc .45

l0lawrence committed:
Merge branch 'typespec_update' of https://github.com/l0lawrence/azure-rest-api-specs into typespec_update

l0lawrence committed:
.45 generated json

l0lawrence committed:
add min-max values

mikeharder committed:
[TypeSpec] Upgrade to 0.45.0 (#24326)

mikeharder committed:
[TypeSpec CI] Enable --warn-as-error (#24340)

- Fixes #24341

raych1 committed:
Added roopesh and armapireview to mgmt plane PR assignees (#24336)

mikeharder committed:
[TypeSpec CI] Pass "--no" to ensure npx does not install anything (#24345)

AvishalomJ committed:
[Hub Generated] Publish private branch 'avjan/new-api-version-for-vasetteings' (#24251)

* Added stable api version for ServerVulnerabilityAssessmentsSettings

* Update system data ref and readme's api-version

* Update ServerVulnerabilityAssessmentsSetting's definition ref

* Updated ServerVulnerabilityAssessmentsSetting's systemData property

* Updated ServerVulnerabilityAssessmentsSetting ref to proxyResource

* Revert "Updated ServerVulnerabilityAssessmentsSetting ref to proxyResource"

This reverts commit f5b7b445b8e7073bc6955ab8c00bde9526330db1.

* Update specification/security/resource-manager/Microsoft.Security/stable/2023-05-01/ServerVulnerabilityAssessmentsSettings.json

Co-authored-by: Zhenhua Hu <zhhu@microsoft.com>

* Update ServerVulnerabilityAssessmentsSettings.json - minor changes

* Update ServerVulnerabilityAssessmentsSettings.json - Added 200 ok response for put request

* Updated ServerVulnerabilityAssessmentsSetting's references to the latest - v5

* Update PutServerVulnerabilityAssessmentsSetting_example.json added 200 ok response

---------

Co-authored-by: Zhenhua Hu <zhhu@microsoft.com>

alzahedi committed:
[Hub Generated] Publish private branch 'dev/alzahedi/update_arc_sql_mi_spec' (#24335)

* Updated spec for arc sql mi to include security and network settings

* Fixed prettier issues

* Resolved spell checks

jingjlii committed:
[Hub Generated] Review request for Microsoft.Batch to add version stable/2023-05-01 (#24163)

poshett committed:
Adding elasticOrganizationToAzureSubscriptionMapping API in new version 2023-06-15-preview for Microsoft.Elastic (#24314)

michaelshikh07 committed:
[Hub Generated] Publish private branch 'azure-kusto/devCM' (#24160)

* [AutoSync] dad295e34cd Merged PR 7593889: Python pipeline: Re-tag python image from pipeline if needed

* [AutoSync] 779e3f273da Merged PR 7983985: NetworkSettingsAllowedFqdnListUpdater: allow updating while suspended

* [AutoSync] 8234bea6386 Merged PR 8021356: Add db pagination to may 2023 version

* [AutoSync] 3a2bf0e47c8 Merged PR 8120727: Region buildout: Fix specifying RpVmSize

* [AutoSync] c87d8729e5a Merged PR 8147686: [Swagger] Fix LintDiff errors

* [AutoSync] 5350cd308a9 Merged PR 8166361: Use upper-case values when configuring CORS

---------

Co-authored-by: swagger-automation <swagger@microsoft.com>

tadelesh committed:
Revert "[Hub Generated] Publish private branch 'azure-kusto/devCM' (#24160)" (#24355)

This reverts commit 3ee139274d100f1e14b443d359d21e1644e297f6.

mikaelmello committed:
Add state and onInactiveMarkAs to Synapse activities (#24075)

Sync ADF changes (https://github.com/Azure/azure-rest-api-specs/pull/23910) to Synapse

l0lawrence committed:
Merge branch 'typespec_update' of https://github.com/l0lawrence/azure-rest-api-specs into typespec_update
